components/match2/wikis/warcraft/legacy/match_group_legacy_convert_map_data.lua

```diff
@@ -0,0 +1,157 @@
+---
+-- @Liquipedia
+-- wiki=warcraft
+-- page=Module:MatchGroup/Legacy/ConvertMapData
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Table = require('Module:Table')
+
+local TBD = 'TBD'
+local NUMBER_OF_OPPONENTS = 2
+
+local ConvertMapData = {}
+
+---@param args table?
+---@return string
+function ConvertMapData.solo(args)
+	args = args or {}
+	--sometimes win param is entered differently ...
+	for _, value, index in Table.iter.pairsByPrefix(args, 'win') do
+		args['map' .. index .. 'win'] = value
+		args['map' .. index] = args['map' .. index] or TBD
+	end
+
+	return Json.stringify(args)
+end
+
+--structure formerly used by `Template:BracketTeamMatch`
+---@param args table?
+---@return string
+function ConvertMapData.team(args)
+	args = args or {}
+
+	local parsedArgs = Table.filterByKey(args, function(key)
+		return string.match(key, 'm%d') == nil
+	end)
+
+	local opponentPlayers = {{}, {}}
+	local mapIndex = 1
+	local prefix = 'm' .. mapIndex
+	while args[prefix .. 'p1'] or args[prefix .. 'p2'] or args[prefix .. 'map'] do
+		local parsedPrefix = 'map' .. mapIndex
+		parsedArgs[parsedPrefix] = args[prefix .. 'map'] or TBD
+		parsedArgs[parsedPrefix .. 'win'] = args[prefix .. 'win']
+
+		for opponentIndex = 1, NUMBER_OF_OPPONENTS do
+			local playerInputPrefix = prefix .. 'p' .. opponentIndex
+			local playerPrefix = parsedPrefix .. 't' .. opponentIndex .. 'p1'
+
+			local name = args[playerInputPrefix .. 'link'] or args[playerInputPrefix] or TBD
+			local player = {
+				name = name,
+				displayname = args[playerInputPrefix] or name,
+				flag = args[playerInputPrefix .. 'flag'],
+				race = args[playerInputPrefix .. 'race'],
+			}
+			opponentPlayers[opponentIndex][name] = player
+
+			parsedArgs[playerPrefix] = name
+			parsedArgs[playerPrefix .. 'race'] = player.race
+			parsedArgs[playerPrefix .. 'heroes'] = args[playerInputPrefix .. 'heroes']
+		end
+
+		mapIndex = mapIndex + 1
+		prefix = 'm' .. mapIndex
+	end
+
+	Array.forEach(opponentPlayers, function(players, opponentIndex)
+		Array.forEach(Array.extractValues(players), function(player, playerIndex)
+			parsedArgs['opponent' .. opponentIndex .. '_p' .. playerIndex] = Json.stringify(player)
+		end)
+	end)
+
+	return Json.stringify(parsedArgs)
+end
+
+--structure formerly used by `Template:BracketTeamMatchMulti`
+---@param args table?
+---@return string
+function ConvertMapData.teamMulti(args)
```

Comment:
Is this not the unsupported ` As of now does NOT support 2v2 legacy matches.` or something else?

Reply:
no 2v2 submatches in team matches =/= 2v2 matches